From 96673e8ec7fb0d78f8caa01477b26b930e5616af Mon Sep 17 00:00:00 2001
From: gvwifi build <build@local>
Date: Sat, 3 Oct 2026 11:31:25 -0400
Subject: [PATCH 3/3] recovery: link libsysutils for libvold

libvold on android-14.1 uses NetlinkEvent (libsysutils); the recovery binary
links libvold statically under TW_INCLUDE_CRYPTO but did not list it.

Change-Id: I459c167e4c42f2e4749e07238ddf4e9b657982f5
---
 Android.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Android.mk b/Android.mk
index ff9296c8..4c1b2a67 100755
--- a/Android.mk
+++ b/Android.mk
@@ -330,6 +330,8 @@ endif
 ifeq ($(TW_INCLUDE_CRYPTO), true)
     LOCAL_CFLAGS += -DTW_INCLUDE_CRYPTO -DUSE_FSCRYPT -Wno-macro-redefined
     LOCAL_SHARED_LIBRARIES += libgpt_twrp
+    # libvold (android-14.1) uses NetlinkEvent from libsysutils
+    LOCAL_SHARED_LIBRARIES += libsysutils
     LOCAL_C_INCLUDES += external/boringssl/src/include bootable/recovery/crypto
     TW_INCLUDE_CRYPTO_FBE := true
     LOCAL_CFLAGS += -DTW_INCLUDE_FBE
